pkg/codegen/nodejs/gen.go

```diff
@@ -1047,45 +1057,103 @@ func (mod *modContext) getNamespaces() map[string]*namespace {
 	}
 
 	for _, t := range mod.types {
-		modName := mod.pkg.TokenToModule(t.Token)
-		if override, ok := mod.modToPkg[modName]; ok {
-			modName = override
-		}
+		modName := mod.getModName(t.Token)
 		ns := getNamespace(modName)
 		ns.types = append(ns.types, t)
 	}
+	for _, e := range mod.enums {
+		modName := mod.getModName(e.Token)
+		ns := getNamespace(modName)
+		ns.enums = append(ns.enums, e)
+	}
 
 	return namespaces
 }
 
-func (mod *modContext) genNamespace(w io.Writer, ns *namespace, input bool, level int) {
+func (mod *modContext) getModName(token string) string {
+	modName := mod.pkg.TokenToModule(token)
+	if override, ok := mod.modToPkg[modName]; ok {
+		modName = override
+	}
+	return modName
+}
+
+func (mod *modContext) genNamespace(w io.Writer, ns *namespace, input, enum bool, level int) {
 	indent := strings.Repeat("    ", level)
 
 	sort.Slice(ns.types, func(i, j int) bool {
 		return tokenToName(ns.types[i].Token) < tokenToName(ns.types[j].Token)
 	})
-	for i, t := range ns.types {
-		if input && mod.details(t).inputType || !input && mod.details(t).outputType {
-			mod.genType(w, t, input, level)
-			if i != len(ns.types)-1 {
+	sort.Slice(ns.enums, func(i, j int) bool {
+		return tokenToName(ns.enums[i].Token) < tokenToName(ns.enums[j].Token)
+	})
+	if enum {
+		for i, e := range ns.enums {
+			mod.genEnum(w, e, level)
+			if i != len(ns.enums)-1 {
 				fmt.Fprintf(w, "\n")
 			}
 		}
+	} else {
+		for i, t := range ns.types {
+			if input && mod.details(t).inputType || !input && mod.details(t).outputType {
+				mod.genType(w, t, input, level)
+				if i != len(ns.types)-1 {
+					fmt.Fprintf(w, "\n")
+				}
+			}
+		}
 	}
 
 	sort.Slice(ns.children, func(i, j int) bool {
 		return ns.children[i].name < ns.children[j].name
 	})
 	for i, child := range ns.children {
 		fmt.Fprintf(w, "%sexport namespace %s {\n", indent, child.name)
-		mod.genNamespace(w, child, input, level+1)
+		mod.genNamespace(w, child, input, enum, level+1)
 		fmt.Fprintf(w, "%s}\n", indent)
 		if i != len(ns.children)-1 {
 			fmt.Fprintf(w, "\n")
 		}
 	}
 }
 
+func (mod *modContext) genEnum(w io.Writer, enum *schema.EnumType, level int) {
+	indent := strings.Repeat("    ", level)
+
+	enumName := tokenToName(enum.Token)
+	for _, e := range enum.Elements {
+		if e.Name == "" {
+			e.Name = makeValidIdentifier(fmt.Sprintf("%v", e.Value))
+		}
+		if e.Comment != "" {
+			fmt.Fprintf(w, "%s/** %s */\n", indent, e.Comment)
+		}
+		fmt.Fprintf(w, "%[1]sexport const %[2]s%[3]s: %[3]s = ", indent, e.Name, enumName)
+		if val, ok := e.Value.(string); ok {
+			fmt.Fprintf(w, "\"%v\";\n", val)
```

Comment:
I think you could do `"%q;\n"` instead
```suggestion
			fmt.Fprintf(w, "%q;\n", val)
```

Reply:
ooh TIL